src/algebra/module/zlattice.lean

```diff
@@ -0,0 +1,224 @@
+    suffices : ‖int.fract (((b.repr) m) i)‖ ≤ 1,
+    { convert mul_le_mul_of_nonneg_right this (norm_nonneg _ : 0 ≤ ‖b i ‖),
+      exact (one_mul _).symm, },
+    rw (norm_one.symm : 1 = ‖(1 : K)‖),
+    apply norm_le_norm_of_abs_le_abs,
+    rw [abs_one, int.abs_fract],
+    exact le_of_lt (int.fract_lt_one _),
+end
+
+section unique
+
+variable [unique ι]
+
+@[simp] lemma coe_floor_self (k : K) : (floor (basis.singleton ι K) k : K) = ⌊k⌋ :=
+basis.ext_elem _ (λ _, by rw [repr_floor_apply, basis.singleton_repr, basis.singleton_repr])
+
+@[simp] lemma coe_fract_self (k : K) : (fract (basis.singleton ι K) k : K) = int.fract k :=
+basis.ext_elem _ (λ _, by rw [repr_fract_apply, basis.singleton_repr, basis.singleton_repr])
+
+end unique
+
+end fintype
+
+lemma fundamental_domain_bounded [finite ι] [has_solid_norm K] :
+  metric.bounded (fundamental_domain b) :=
+begin
+  casesI nonempty_fintype ι,
+  use 2 * ∑ j, ‖b j‖,
+  intros x hx y hy,
+  refine le_trans (dist_le_norm_add_norm x y) _,
+  rw [← fract_eq_self.mpr hx, ← fract_eq_self.mpr hy],
+  refine (add_le_add (norm_fract_le b x) (norm_fract_le b y)).trans _,
+  rw ← two_mul,
+end
+
+lemma vadd_mem_fundamental_domain [fintype ι] (y : span ℤ (set.range b)) (x : E) :
+  y +ᵥ x ∈ fundamental_domain b ↔ y = -floor b x :=
+by rw [subtype.ext_iff, ← add_right_inj x, add_subgroup_class.coe_neg, ← sub_eq_add_neg,
+    ← fract_apply, ← fract_zspan_add b _ (subtype.mem y), add_comm, ← vadd_eq_add, ← vadd_def,
+    eq_comm, ← fract_eq_self]
+
+lemma exist_unique_vadd_mem_fundamental_domain [finite ι] (x : E) :
+  ∃! v : span ℤ (set.range b), v +ᵥ x ∈ fundamental_domain b :=
+begin
+  casesI nonempty_fintype ι,
+  refine ⟨-floor b x, _, λ y h, _⟩,
+  { exact (vadd_mem_fundamental_domain b (-floor b x) x).mpr rfl, },
+  { exact (vadd_mem_fundamental_domain b y x).mp h, },
+end
+
+end normed_lattice_field
+
+section real
+
+variables [normed_add_comm_group E] [normed_space ℝ E]
+variables (b : basis ι ℝ E)
+
+@[measurability]
+lemma fundamental_domain_measurable_set [measurable_space E] [opens_measurable_space E]
+  [finite ι] :
+  measurable_set (fundamental_domain b) :=
+begin
+  haveI : finite_dimensional ℝ E := finite_dimensional.of_fintype_basis b,
+  let f := (finsupp.linear_equiv_fun_on_finite ℝ ℝ ι).to_linear_map.comp b.repr.to_linear_map,
+  let D : set (ι → ℝ) := set.pi set.univ (λ i : ι, (set.Ico (0 : ℝ) 1)),
+  rw ( _ : fundamental_domain b = f⁻¹' D),
+  { refine measurable_set_preimage (linear_map.continuous_of_finite_dimensional f).measurable _,
+    exact pi set.univ.to_countable (λ (i : ι) (H : i ∈ set.univ), measurable_set_Ico), },
+  { ext,
+    simp only [fundamental_domain, set.mem_set_of_eq, linear_map.coe_comp,
+      linear_equiv.coe_to_linear_map, set.mem_preimage, function.comp_app, set.mem_univ_pi,
+      finsupp.linear_equiv_fun_on_finite_apply], },
+end
+
+protected lemma is_add_fundamental_domain [finite ι] [measurable_space E]
```

Comment:
Why `protected`?

Reply:
To avoid conflicting with `measure_theory.is_add_fundamental_domain`, the actual predicate (vs proofs of such).